### PR TITLE
doq: init at 0.9.1

### LINKED ISSUE #241658: Package request: doq
**Project description**

python docstring generator

**Metadata**

* homepage URL: https://github.com/heavenshell/py-doq
* source URL: https://github.com/heavenshell/py-doq
* license: bsd
* platforms: unix, linux, darwin, ...

### PR DESCRIPTION
**Pull request by natsukium**
###### Description of changes

doq is a tiny tool to generate python's docstring.
https://github.com/heavenshell/py-doq

close #201155, close #241658

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/doq/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "doq";
+  version = "0.9.1";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "heavenshell";
+    repo = "py-doq";
+    rev = "refs/tags/${version}";
```
Comment (drupol) [inline pkgs/development/tools/misc/doq/default.nix:14]:
```suggestion
    rev = version;
```

File: pkgs/development/tools/misc/doq/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  nativeCheckInputs = with python3.pkgs; [
+    parameterized
+    unittestCheckHook
+  ];
+
+  pythonImportsCheck = [ "doq" ];
+
+  meta = with lib; {
+    description = "Docstring generator";
```
Comment (drupol) [inline pkgs/development/tools/misc/doq/default.nix:32]: The description is definitely too succinct.

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/misc/doq/default.nix (hunk 1 above)
Comment (natsukium) [reply pkgs/development/tools/misc/doq/default.nix:14]: I like to specify "refs/tags" because some repositories have branches with names like versions.

Comment (drupol) [reply pkgs/development/tools/misc/doq/default.nix:14]: Ok fair enough.

File: pkgs/development/tools/misc/doq/default.nix (hunk 2 above)
Comment (natsukium) [reply pkgs/development/tools/misc/doq/default.nix:32]: This is the only description I could find on the upstream initially.
How about `Docstring generator for python`?
We can tell it's a python application. If you know python, you understand the word docstring.

Comment (drupol) [reply pkgs/development/tools/misc/doq/default.nix:32]: That's the thing, I've never use python, that's the reason why I found the description a bit too vague.

**Review by drupol — APPROVED** [review]

**Pull request merged**